modules/search-service-impl/src/main/resources/OSGI-INF/search-service-rest.xml

```diff
@@ -15,4 +15,6 @@
              cardinality="1..1" policy="static" bind="setSearchService"/>
   <reference name="serviceregistry" interface="org.opencastproject.serviceregistry.api.ServiceRegistry"
              cardinality="1..1" policy="static" bind="setServiceRegistry"/>
+  <reference name="seriesservice" interface="org.opencastproject.series.api.SeriesService"
+             bind="setSeriesService"/>
```

Comment:
The REST endpoint assumes the series service exists and that this cardinality is 1..1. IMHO, it's human friendly to clarify that requirement in this doc.

HOWEVER, as I mentioned in early comment, tying the Engage Search service to the Admin's Series service, for sites that use the SeriesServiceRemote on Engage, can/will cause service interruptions for students watching lectures that was not expected. It would be more site friendly to make this dependency dynamic and optional and change the REST code to return an unimplemented 501, or similar if the sname is passed in. NOTE that the seriesId param uses the Search Service series path. If a series has no events published to engage, the seriesId query returns the no-results response. It's odd for the sname to return different results than sid.

Reply:
First of all, thanks for this feedback "from the trenches". I hadn't thought of the problems of tying the two services together.

First, regarding the response in case the series doesn't exist. You're right, with a series ID instead of a name, this gives an empty result. I have now adapted the function to do that.

Second, regarding the dynamic dependency: I have now declared the series service `dynamic` and of cardinality `0..1`. Also, the method returns 501 as you suggested (I thought about 503 instead, but couldn't make up my mind), so you don't pay a price if you're not using the feature. Since I'm on OSGi god, I'm hoping this is enough. The code seems to be working.